### PR TITLE
[WIP] Install and used Devise, Cancancan, Rails admin gem

### PR DESCRIPTION
**Pull request by bien-thy-mulodo**
Research about gems:
- [Devise](https://github.com/heartcombo/devise): Flexible authentication solution for Rails with Warden
- [Cancancan](https://github.com/CanCanCommunity/cancancan): The authorization Gem for Ruby on Rails.
- [Rails admin](https://github.com/sferik/rails_admin): RailsAdmin is a Rails engine that provides an easy-to-use interface for managing your data

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
```
Comment (nguyen-dat-mulodo) [inline app/models/ability.rb]: @bien-thy-mulodo At admin page, if user is not authorized, The error from system is showing. Please create a page to show it.

Comment (bien-thy-mulodo) [reply app/models/ability.rb]: I updated, please check it again.
Thank you!

File: app/controllers/users/confirmations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::ConfirmationsController < Devise::ConfirmationsController
+  # GET /resource/confirmation/new
```
Comment (nguyen-dat-mulodo) [inline app/controllers/users/confirmations_controller.rb]: @bien-thy-mulodo Please remove unused code. Thanks.

File: app/models/category.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class Category < ApplicationRecord
+  has_many :posts
```
Comment (nguyen-dat-mulodo) [inline app/models/category.rb]: @bien-thy-mulodo Models `Category`, `Post` are not using at anywhere. Please check it again.

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<h1>Posts</h1>
+<% if current_user.nil? %>
+  <%= link_to new_user_session_path, class: 'login-button' do %>Sign in<% end %>
+<% else %>
```
Comment (nguyen-dat-mulodo) [inline app/views/posts/index.html.erb]: @bien-thy-mulodo You can check authorization `edit` with cancancan 
Ex:
```
<% if can? :write, current_user %>
  <%= link_to('Edit registration', edit_user_registration_path) %>
<% end %>
```

File: app/views/users/registrations/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<h2>Sign up</h2>
+
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
```
Comment (nguyen-dat-mulodo) [inline app/views/users/registrations/new.html.erb]: @bien-thy-mulodo Folder `devise` not exists. Please check it again.

**Comment by nguyen-dat-mulodo** [comment]
@bien-thy-mulodo I merged branch `feature/dotenv-gem` to develop. Please check and solve conflict. Thanks.

File: app/models/category.rb (hunk 3 above)
Comment (bien-thy-mulodo) [reply app/models/category.rb]: @nguyen-dat-mulodo Post model, Category model used for Rails admin gem.

File: app/views/posts/index.html.erb (hunk 4 above)
Comment (bien-thy-mulodo) [reply app/views/posts/index.html.erb]: @nguyen-dat-mulodo I didn't use this file so I will delete it.

**Comment by bien-thy-mulodo** [comment]
@nguyen-dat-mulodo 
I updated. Please check it again.
Thank you!

**Pull request merged**